src/storage/mvcc/reader/reader.rs

```diff
@@ -77,6 +77,13 @@ pub struct MvccReader<S: Snapshot> {
     scan_mode: Option<ScanMode>,
     key_only: bool,
 
+    /// True means following operations are performed on keys with the same prefix, i.e.,
+    /// different versions of the same key. It can use prefix seek to speed up reads from write-cf.
+    /// The `scan_mode` is ignored if it's true.
+    single_key: bool,
```

Comment:
Can we assume we are using the `single_key` mode when `scan_mode` is `None`? Then maybe we don't need this field?

Reply:
OK. Only support `single_key` mode and scan mode.

Reply:
Is it confirmed that we can reuse the reader on different keys even if single_key is set to true?